app/helpers/application_helper.rb

```diff
@@ -54,6 +54,15 @@ def render_index_grid(grid)
     render "admin/common/grid", no_records: no_records, grid: grid
   end
 
+  def render_loan_grid(grid)
+    if grid.all_pages_records.count == 1 && grid.filtered_by == ['id']
```

Comment:
Yes you should generalize. Otherwise it doesn't belong in application_helper.

You should be able to do `redirect_to grid.all_pages_records.first` right? Then it doesn't have to mention `loan` at all? You could just call it `render_index_grid_with_redirect_check` or something.

BUT

Now that I think of it, it seems pretty strange to be redirecting in a helper. I can imagine my future self tearing my hair out trying to figure out why this action is redirecting. This really belongs in the controller. Please find a way to do it there. e.g. add a method called `redirect_if_one_match_and_id_filter(relation)` to ApplicationController and call it right after the `filter_by_params` in the LoansController, passing `@loans`.